bin/prover-client/src/config.rs

```diff
@@ -0,0 +1,272 @@
+    #[serde(default)]
+    pub(crate) rpc: RpcConfig,
+
+    /// Worker configuration for different proving backends.
+    #[serde(default)]
+    pub(crate) workers: WorkerConfig,
+
+    /// Polling and timing configuration.
+    #[serde(default)]
+    pub(crate) timing: TimingConfig,
+
+    /// Retry policy configuration.
+    #[serde(default)]
+    pub(crate) retry: RetryConfig,
+
+    /// Feature flags.  
+    #[serde(default)]
+    pub(crate) features: FeatureConfig,
+}
+
+/// RPC configuration for the prover client.
+#[derive(Debug, Clone, Serialize, Deserialize, Default)]
+pub(crate) struct RpcConfig {
+    /// The JSON-RPC port for development mode.
+    #[serde(default = "default_values::dev_rpc_port")]
+    pub dev_port: usize,
+
+    /// The base URL for JSON-RPC endpoint in development mode
+    #[serde(default = "default_values::dev_rpc_url")]
+    pub dev_url: String,
+}
+
+/// Worker configuration for the prover client.
+#[derive(Debug, Clone, Serialize, Deserialize, Default)]
+pub(crate) struct WorkerConfig {
+    /// Number of native prover workers.
+    #[serde(default = "default_values::workers")]
+    pub(crate) native: usize,
+
+    /// Number of SP1 prover workers.
+    #[serde(default = "default_values::workers")]
+    pub(crate) sp1: usize,
+
+    /// Number of Risc0 prover workers.
+    #[serde(default = "default_values::workers")]
+    pub(crate) risc0: usize,
+}
+
+/// Timing configuration for the prover client.
+#[derive(Debug, Clone, Serialize, Deserialize, Default)]
+pub(crate) struct TimingConfig {
+    /// Polling interval for prover manager loop in milliseconds.
+    #[serde(default = "default_values::polling_interval_ms")]
+    pub(crate) polling_interval_ms: u64,
+
+    /// Checkpoint polling interval in seconds.
+    #[serde(default = "default_values::checkpoint_poll_interval_s")]
+    pub(crate) checkpoint_poll_interval_s: u64,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, Default)]
+pub(crate) struct RetryConfig {
+    /// Maximum number of retries for transient failures.
+    #[serde(default = "default_values::max_retry_counter")]
+    pub(crate) max_retry_counter: u64,
+
+    /// Default number of Bitcoin RPC retries.
+    #[serde(default = "default_values::bitcoin_retry_count")]
+    pub(crate) bitcoin_retry_count: u8,
+
+    /// Default Bitcoin RPC retry interval in milliseconds.
+    #[serde(default = "default_values::bitcoin_retry_interval_ms")]
+    pub(crate) bitcoin_retry_interval_ms: u64,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize, Default)]
+pub(crate) struct FeatureConfig {
+    /// Enable development RPC endpoints.
+    #[serde(default = "default_values::enable_dev_rpcs")]
+    pub(crate) enable_dev_rpcs: bool,
+
+    /// Enable checkpoint proof runner.
+    #[serde(default = "default_values::enable_checkpoint_runner")]
+    pub(crate) enable_checkpoint_runner: bool,
+}
+
+/// Default value functions to make [`serde`] happy and make the bloody [`super`] code mess easy to
```

Comment:
I guess we don't want such words in our repo.
```suggestion
/// Default value functions to make [`serde`] happy and make the [`super`] code mess easy to
```